pkg/pillar/hypervisor/xen.go

```diff
@@ -488,7 +488,10 @@ func (ctx xenContext) LookupByName(domainName string, domainID int) (int, error)
 		log.Warningf("domainid changed from %d to %d for %s\n",
 			domainID, domainID2, domainName)
 	}
-	return domainID2, err
+	if !isDomainRunning(domainID2) { //check if domain is in healthy state
+		return domainID2, fmt.Errorf("xl domain %s not running", domainName)
+	}
+	return domainID2, nil
```

Comment:
Cool. That's a good place to put it. 
 But the comment might be misleading. It is about shutting down hence pretending it is already gone, not about it being unhealthy.

Reply:
Changed.